tests/logprob/test_transforms.py

```diff
@@ -1159,6 +1160,73 @@ def test_special_log_exp_transforms(transform):
         assert equal_computations([logp_test], [logp_ref])
 
 
+def test_power_const_exponent_output():
+    x_rv_pow = pt.pow(2, pt.random.normal())
+    x_rv_exp2 = pt.exp2(pt.random.normal())
+
+    x_vv_pow = x_rv_pow.clone()
+    x_vv_exp2 = x_rv_exp2.clone()
+
+    x_logp_fn_pow = pytensor.function([x_vv_pow], pt.sum(logp(x_rv_pow, x_vv_pow)))
+    x_logp_fn_exp2 = pytensor.function([x_vv_exp2], pt.sum(logp(x_rv_exp2, x_vv_exp2)))
+
+    np.testing.assert_allclose(x_logp_fn_pow(0.1), x_logp_fn_exp2(0.1))
+
+
+def test_power_const_exponent_negative_rv_error():
+    # test when const < 0 we raise error
+    # test with RV when logp(<0) we raise error
+    base_rv = pt.random.normal([2])
+    x_raw_rv = pt.random.normal()
+    x_rv = pt.power(base_rv, x_raw_rv)
+
+    x_rv.name = "x"
+    base_rv.name = "base"
+    base_vv = base_rv.clone()
+    x_vv = x_rv.clone()
+
+    res = conditional_logp({base_rv: base_vv, x_rv: x_vv})
+    factors = [factor for factor in res.values()]
+    logp_vals_fn = pytensor.function([base_vv, x_vv], factors[1])
+
+    with pytest.raises(ParameterValueError, match="base >= 0"):
+        logp_vals_fn(np.array([-2]), np.array([2]))
+
+
+def test_power_const_exponent_negative_const_error():
+    with pytest.raises(ParameterValueError, match="base >= 0"):
+        x_rv = pt.pow(-2, pt.random.normal())
+        x_vv = x_rv.clone()
+        logp(x_rv, x_vv)
```

Comment:
You can test this in the constant test, no need to rederive the logp in a separate test (should be a bit faster, and it's conceptually related anyway)

Reply:
Not sure I understand. I've changed the test to be,
```python
def test_measurable_power_exponent_with_constant_base():
    x_rv_pow = pt.pow(2, pt.random.normal())
    x_rv_exp2 = pt.exp2(pt.random.normal())

    x_vv_pow = x_rv_pow.clone()
    x_vv_exp2 = x_rv_exp2.clone()

    x_logp_fn_pow = pytensor.function([x_vv_pow], pt.sum(logp(x_rv_pow, x_vv_pow)))
    x_logp_fn_exp2 = pytensor.function([x_vv_exp2], pt.sum(logp(x_rv_exp2, x_vv_exp2)))

    np.testing.assert_allclose(x_logp_fn_pow(0.1), x_logp_fn_exp2(0.1))

    with pytest.raises(ParameterValueError, match="base >= 0"):
        x_rv_neg = pt.pow(-2, pt.random.normal())
        x_vv_neg = x_rv_neg.clone()
        logp(x_rv_neg, x_vv_neg)
```

But I don't think thats what you want, as I'm still rederiving the logp.

Reply:
Hmm so the Assert is actually triggered before the `logp` is ever evaluated? Okay, I guess we can let it be.